src/include/storage/storage_util.h

```diff
@@ -189,13 +189,52 @@ class StorageUtil {
                                               static_cast<uint32_t>(str.size()));
   }
 
+  /**
+   * Helper method to serialize a vector of strings into a VarlenEntry
+   * @param vec input whose elements are to be serialized into a VarlenEntry
+   * @return varlen entry representing this data vector
+   * @warning checking IsInlined() to see if you need to possibly clean up a buffer
+   * @warning the length of the data vector is not serialized
+   */
+  static storage::VarlenEntry CreateVarlen(const std::vector<const std::string> &vec) {
```

Comment:
We probably want to rename this to `CreateArray` since we don't want everyone to use this.
Another issue is that it only supports `std::string`. What if it took in a vector of `TransientValue`?

Reply:
I think we should avoid transient values since that was a major source of complexity in the previous catalog and we should always know a column type in the catalog at compile time.  Maybe we have a group of helpers that are Create[Type]Array?

Reply:
@apavlo For now I made a template function for other types below that actually for other fixed size types.